library/src/main/java/com/joom/grip/io/JrtFileSource.kt

```diff
@@ -0,0 +1,21 @@
+package com.joom.grip.io
+
+import java.nio.file.Files
+import java.nio.file.Path
+import kotlin.io.path.extension
+
+class JrtFileSource(private val jrtPath: Path) : FileSource {
+
+  override fun listFiles(callback: (name: String, type: FileSource.EntryType) -> Unit) {
+    Files.walk(jrtPath)
+      .filter { it.extension.endsWith("class") }
+      .forEach { callback(it.subpath(jrtPath.nameCount, it.nameCount).toString(), FileSource.EntryType.CLASS) }
```

Comment:
Shouldn't it be `it.relativize(jrtPath)`?

Reply:
Well `jrtPath.relativize(it)` actually, but I got the idea